### PR TITLE
harddisk templates: organize by vendors

### PR DESCRIPTION
**Pull request by jouvin**
- Required by Aquilon
- Previous templates kept but deprecated
- Generic templates under 'generic' vendor

### CODE REVIEW

File: hardware/harddisk/hp_smartarray.pan [hunk 1]
```diff
@@ -1,12 +1,5 @@
...
-############################################################
-
 structure template hardware/harddisk/hp_smartarray;
 
-"model"     = "SAS Raid Controller";
-"capacity"  = undef;
-"interface" = "sas";
-"part_prefix" = 'p';
+include 'hardware/harddisk/hp/smartarray';
+include 'hardware/hardidisk/deprecation-warning';
```
Comment (jrha) [inline hardware/harddisk/hp_smartarray.pan]: hard**i**disk?

**Comment by jrha** [comment]
When did this become required by aquilon? We haven't done this.

File: hardware/harddisk/hp_smartarray.pan (hunk 1 above)
Comment (jouvin) [reply hardware/harddisk/hp_smartarray.pan]: Thanks for spotting the typo!

File: hardware/harddisk/generic/ide.pan [hunk 2]
```diff
@@ -0,0 +1,11 @@
+############################################################
+#
+# RESPONSIBLE: Charles Loomis <charles.loomis@cern.ch>
+#
+############################################################
+
+structure template hardware/harddisk/ide;
```
Comment (jrha) [inline hardware/harddisk/generic/ide.pan]: harddisk/**generic**/ide

File: hardware/harddisk/generic/ide.pan [hunk 3]
```diff
@@ -0,0 +1,11 @@
+############################################################
+#
+# RESPONSIBLE: Charles Loomis <charles.loomis@cern.ch>
+#
+############################################################
```
Comment (jrha) [inline hardware/harddisk/generic/ide.pan]: Can we drop these comment blocks? They're wrong and basically meaningless.

**Comment by jouvin** [comment]
@jrha sorry I overlooked the IDE-related templates. The reason for the wrong headers (a very bad idea from Cal!) and the wrong template namespace. Should be fixed now.

File: hardware/harddisk/deprecation-warning.pan [hunk 4]
```diff
@@ -0,0 +1,9 @@
+structure template hardware/harddisk/deprecation-warning;
+
+'type' = {
+    deprecated(0,
+                'Use of hardisk templates immediately under hardware/harddisk/ is deprecated. ' +
+                'Please use templates under hardware/harddisk/disk_manufacturer/ ' +
+                'or hardware/harddisk/generic/ instead.');
+    'deprecated'
+};
```
Comment (jrha) [inline hardware/harddisk/deprecation-warning.pan]: If you want to fix this:
```pan
'type' = {
    deprecated(0,
        'Use of hardisk templates immediately under hardware/harddisk/ is deprecated. ' +
        'Please use templates under hardware/harddisk/disk_manufacturer/ ' +
        'or hardware/harddisk/generic/ instead.'
    );
    'deprecated'
};
```

**Comment by jouvin** [comment]
@jrha I managed to get panlint happy 😄

**Comment by jrha** [comment]
Cool. The aim is not to make panlint happy though, it is to write readable code 😁

**Comment by jouvin** [comment]
Agreed... but the most readable for me was something making panlint unhappy because of the very long line or the identation not aligned on a multiple of 4 spaces!

**Comment by jrha** [comment]
In that case feel free to ignore the error

**Pull request merged**